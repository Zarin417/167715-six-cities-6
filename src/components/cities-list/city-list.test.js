import React from "react";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {createMemoryHistory} from "history";
import {Provider} from "react-redux";
import {Router} from "react-router";
import CitiesList from "./cities-list";

const mockStore = configureStore({});
const cities = [`London`, `Paris`, `Berlin`, `Odence`, `Venice`, `Copenhagen`];

it(`CitiesList should render correctly`, () => {
  const history = createMemoryHistory();
  const store = mockStore({
    OFFER: {activeCity: `Copenhagen`}
  });
  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <CitiesList
            cities={cities}
          />
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
