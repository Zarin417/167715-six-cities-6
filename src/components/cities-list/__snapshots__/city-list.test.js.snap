// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList should render correctly 1`] = `
<div>
  <ul
    class="locations__list tabs__list"
  >
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item "
        href="/city/London"
      >
        <span>
          London
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item "
        href="/city/Paris"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item "
        href="/city/Berlin"
      >
        <span>
          Berlin
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item "
        href="/city/Odence"
      >
        <span>
          Odence
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item "
        href="/city/Venice"
      >
        <span>
          Venice
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item tabs__item--active"
        href="/city/Copenhagen"
      >
        <span>
          Copenhagen
        </span>
      </a>
    </li>
  </ul>
</div>
`;
